File: 3490-find-the-maximum-length-of-valid-subsequence-i/3490-find-the-maximum-length-of-valid-subsequence-i.cpp
class Solution {
public:
    int maximumLength(vector<int>& nums) {
        int prev = nums[0];
        int alternating = 1;
        int even = 0;
        int odd = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] % 2 != prev % 2)
            {
                alternating++;
                prev = nums[i];
            }
            if(nums[i] % 2 == 0)
            {
                even++;
            }
            else
            {
                odd++;
            }
        } 

        return max({alternating, even, odd});
    }
};

File: 3490-find-the-maximum-length-of-valid-subsequence-i/3490-find-the-maximum-length-of-valid-subsequence-i.py
class Solution:
    def maximumLength(self, nums: List[int]) -> int:
        prev = nums[0]
        alternating = 1
        even = 0
        odd = 0
        for num in nums:
            if num % 2 != prev % 2:
                alternating += 1
                prev = num
            if num % 2 == 0:
                even += 1
            else:
                odd += 1

        return max(alternating, even, odd)


        